jscomp/runtime/js_array.ml

```diff
@@ -25,216 +25,193 @@
 (** JavaScript Array API *)
 
 type 'a t = 'a array
-type 'a array_like = 'a Js_array2.array_like
+type 'a array_like
 
-(* commented out until bs has a plan for iterators
+(* commented out until Melange has a plan for iterators
    type 'a array_iter = 'a array_like
 *)
 
 external from : 'a array_like -> 'a array = "Array.from"
 (* ES2015 *)
 
-external fromMap : 'a array_like -> (('a -> 'b)[@mel.uncurry]) -> 'b array
+external fromMap : 'a array_like -> f:(('a -> 'b)[@mel.uncurry]) -> 'b array
   = "Array.from"
-
 (* ES2015 *)
 
 external isArray : 'a -> bool = "Array.isArray"
-
-(* ES2015 *)
 (* ES2015 *)
 
 (* Array.of: seems pointless unless you can bind *)
 external length : 'a array -> int = "length" [@@mel.get]
 
-(* Mutator functions
-*)
-external copyWithin : to_:int -> 'this = "copyWithin"
-[@@mel.send.pipe: 'a t as 'this]
+(** Mutating functions *)
+
+external copyWithin : 'a t -> to_:int -> 'a t = "copyWithin" [@@mel.send]
 (* ES2015 *)
 
-external copyWithinFrom : to_:int -> from:int -> 'this = "copyWithin"
-[@@mel.send.pipe: 'a t as 'this]
+external copyWithinFrom : 'a t -> to_:int -> from:int -> 'a t = "copyWithin"
+[@@mel.send]
```

Comment:
I wonder if we apply this approach across the board for melange bindings, maybe we can get rid of `send.pipe` eventually? I recall Bob mentioning implementing it added a lot of complexity (plus, from user point of view having two ways to do the same thing is confusing).

Reply:
This probably warrants another discussion.

Here's the [PR](https://github.com/rescript-lang/rescript-compiler/pull/5146/files?w=1) where `send.pipe` got removed from ReScript. I don't think it's too much code to maintain, personally.